### DISCUSSION TITLE
Allow usage of a plugin instance for Player#setAllowFlight(), like with Player#hideEntity()

### DISCUSSION THREAD

**Discussion by cometcake575** [PaperMC/Paper#10489] [Ideas]
### Is your feature request related to a problem?

In my plugin I sometimes use `Player#setAllowFlight()` for a few abilities, like allowing players in cobwebs to climb around. However this sometimes causes conflicts with other plugins that add features like /fly.

### Describe the solution you'd like.

Similar to the method `Player#hideEntity(Plugin, Entity)`, the setAllowFlight method could also take a plugin as a parameter, enabling flight for a player if at least one plugin enables flight.

This means a /fly plugin could set flying to true, and my plugin could disable flying as it is not needed for any abilities, and the player would still be able to fly.

### Describe alternatives you've considered.

I implemented the /fly command with a permission into my own plugin, giving it compatibility with the plugin's features, however other plugins may still use flight for some things so it would work better if they could individually set flight in a way that doesn't interfere with other plugins.

**Comment by Machine-Maker** [comment]
I don't think this is a good idea. One can make this same argument for pretty much every "toggle" in the game, and that isn't really a good idea. What I would do, is track if the player had flight before you give it to them, and then don't remove the flight when your few seconds ends if they had it before. 

Just simply adding what you propose here won't solve the issue, because plugins then have to actually *use* it. So instead of just updating your plugin, all plugins would have to update.
